packages/plugins/experimental/plugin-script/src/components/ScriptSettingsPanel.tsx

```diff
@@ -0,0 +1,107 @@
+//
+// Copyright 2024 DXOS.org
+//
+
+import React, { ChangeEvent, KeyboardEvent, useCallback } from 'react';
+
+import { useClient } from '@dxos/react-client';
+import { Filter, getMeta, getSpace, useQuery } from '@dxos/react-client/echo';
+import { Button, Icon, Input, useControlledValue, useTranslation } from '@dxos/react-ui';
+
+import { SCRIPT_PLUGIN } from '../meta';
+import { FunctionType, type ScriptType } from '../types';
+import { getInvocationUrl, getUserFunctionUrlInMetadata } from '../edge';
+
+export type ScriptSettingsPanelProps = {
+  script: ScriptType;
+};
+
+export const ScriptSettingsPanel = ({ script }: ScriptSettingsPanelProps) => {
+  const { t } = useTranslation(SCRIPT_PLUGIN);
+  const client = useClient();
+  const space = getSpace(script);
+  // TODO(dmaretskyi): Parametric query.
+  const [fn] = useQuery(
+    space,
+    Filter.schema(FunctionType, (fn) => fn.source === script),
+  );
+
+  const handleNameChange = useCallback(
+    (event: ChangeEvent<HTMLInputElement>) => {
+      script.name = event.target.value;
+    },
+    [script],
+  );
+
+  const [binding, setBinding] = useControlledValue(fn?.binding ?? '');
+  const handleBindingChange = useCallback(
+    (event: ChangeEvent<HTMLInputElement>) => {
+      setBinding(event.target.value);
+    },
+    [setBinding],
+  );
+  const handleBindingBlur = useCallback(() => {
+    fn.binding = binding;
+  }, [fn, binding]);
+
+  const functionPath = fn && getUserFunctionUrlInMetadata(getMeta(fn));
+  const functionUrl =
+    functionPath &&
+    getInvocationUrl(functionPath, client.config.values.runtime?.services?.edge?.url ?? '', {
+      spaceId: space?.id,
+    });
+
+  // TODO(wittjosiah): Use ClipboardProvider.
+  const handleCopy = useCallback(() => {
+    functionUrl && navigator.clipboard.writeText(functionUrl);
+  }, [functionUrl]);
+
+  return (
+    <div role='form' className='flex flex-col w-full p-2 gap-4'>
```

Comment:
use DeprecatedFormInput?
(in common with other settings so that we can easily replace these all later?)

Reply:
that lays things out in a way that doesn't work for the sidebar